shared/enclave/server/wallets/ledger.ts

```diff
@@ -0,0 +1,85 @@
+import EthTx from 'ethereumjs-tx';
+import { addHexPrefix, toBuffer } from 'ethereumjs-util';
+import { WalletLib } from 'shared/enclave/types';
+import TransportNodeHid from '@ledgerhq/hw-transport-node-hid';
+import LedgerEth from '@ledgerhq/hw-app-eth';
+let transport: any;
+
+async function getEthApp() {
+  try {
+    if (!transport) {
+      transport = await TransportNodeHid.create();
+      transport.on('disconnect', () => (transport = null));
+    }
+    return new LedgerEth(transport);
+  } catch (err) {
+    if (err && err.message && err.message.includes('cannot open device with path')) {
+      throw new Error(
+        'Your Ledger is currently in use with another application. Please wait, or close other wallet applications before trying again.'
```

Comment:
Should have this i18n'd i think since its a user error

Reply:
Absolutely, though I have no way of getting at redux state in here. Probably what I should do is replace all of these human readable errors with ERROR_CODES to be `translate`'d on the frontend.